boards/opentitan/src/main.rs

```diff
@@ -457,6 +463,36 @@ unsafe fn setup() -> (
         static _estorage: u8;
     }
 
+    // Flash setup memory protection for the ROM/Kernel
+    let etext_addr = &_etext as *const u8 as u32;
+    let smanifest_addr = &_manifest as *const u8 as u32;
+    // Protect from ROM start to end of text
+    let text_end = etext_addr.checked_sub(smanifest_addr).unwrap(); // Unwrap fail = addresses have updated in an unexpected way, this should be revised.
+    let num_pages_occupied = div_ceil(text_end, lowrisc::flash_ctrl::PAGE_SIZE as u32);
+    // Only allow reads for this region, any other ops will cause an MP fault
+    let mp_cfg = FlashMPConfig {
+        read_en: true,
+        write_en: false,
+        erase_en: false,
+        scramble_en: false,
+        ecc_en: false,
+        he_en: false,
+    };
+
+    // Starting from page 0 (start of flash) to num_pages_occupied on `region 0`.
+    if let Err(e) =
+        peripherals
+            .flash_ctrl
+            .mp_set_region_perms(0, num_pages_occupied as usize, 0, &mp_cfg)
```

Comment:
I feel like this API should match the MPU/PMP API. So in `main()` we can just pass in the same addresses and handle the conversion to pages in the flash controller. Especially as this isn't a generic HIL but an OT specific API

Reply:
sounds good, I'm thinking this:
```
peripherals
            .flash_ctrl
            .mp_set_region_perms(start_addr, end_addr, region_num, &mp_cfg)
```
and in `pub fn mp_set_region_perms()`, we check if the addresses are valid then convert to pages with a helper function to calculate the num pages between `start_addr & end_addr`.

Does that sound okay?

Reply:
That sounds good!

The only question left is do we need region number?

Reply:
cool! hmm... in this instance, we don't I suppose, since we are just setting up the single region. But later down the line could be useful if we wanted to have multiple isolated regions with varying levels of access control.

Reply:
Ok, might be best to provide a bit more info in the doc comments about what a region number is